ledger/eval_simple_test.go

```diff
@@ -263,46 +263,73 @@ func TestPayoutFees(t *testing.T) {
 		dl.txns(&pay, pay.Args("again"))
 		vb := dl.endBlock(proposer)
 
-		const bonus1 = 5_000_000 // the first bonus value, set in config/consensus.go
+		postsink := micros(dl.t, dl.generator, genBalances.FeeSink)
+		postprop := micros(dl.t, dl.generator, proposer)
+		t.Log(" postsink", postsink)
+		t.Log(" postprop", postprop)
+
+		prp = lookup(dl.t, dl.generator, proposer)
+
+		const bonus1 = 10_000_000 // the first bonus value, set in config/consensus.go
 		if ver >= payoutsBegin {
 			require.True(t, dl.generator.GenesisProto().Payouts.Enabled)    // version sanity check
 			require.NotZero(t, dl.generator.GenesisProto().Payouts.Percent) // version sanity check
 			// new fields are in the header
 			require.EqualValues(t, 2000, vb.Block().FeesCollected.Raw)
 			require.EqualValues(t, bonus1, vb.Block().Bonus.Raw)
 			require.EqualValues(t, bonus1+1_500, vb.Block().ProposerPayout().Raw)
+			// This last one is really only testing the "fake" agreement that
+			// happens in dl.endBlock().
+			require.EqualValues(t, proposer, vb.Block().Proposer())
+
+			// At the end of the block, part of the fees + bonus have been moved to
+			// the proposer.
+			require.EqualValues(t, bonus1+1500, postprop-preprop) // based on 75% in config/consensus.go
+			require.EqualValues(t, bonus1-500, presink-postsink)
+			require.Equal(t, prp.LastProposed, dl.generator.Latest())
 		} else {
 			require.False(t, dl.generator.GenesisProto().Payouts.Enabled)
 			require.Zero(t, dl.generator.GenesisProto().Payouts.Percent) // version sanity check
 			require.Zero(t, vb.Block().FeesCollected)
 			require.Zero(t, vb.Block().Bonus)
 			require.Zero(t, vb.Block().ProposerPayout())
+			// fees stayed in the feesink
+			require.EqualValues(t, 0, postprop-preprop, "%v", proposer)
+			require.EqualValues(t, 2000, postsink-presink)
+			require.Zero(t, prp.LastProposed)
 		}
 
-		postsink := micros(dl.t, dl.generator, genBalances.FeeSink)
-		postprop := micros(dl.t, dl.generator, proposer)
-		t.Log(" postsink", postsink)
-		t.Log(" postprop", postprop)
+		// Do another block, make sure proposer doesn't get paid again. (Sanity
+		// check. The code used to award the payout used to be in block n+1).
+		vb = dl.fullBlock()
+		require.Equal(t, postsink, micros(dl.t, dl.generator, genBalances.FeeSink))
+		require.Equal(t, postprop, micros(dl.t, dl.generator, proposer))
 
-		// At the end of the block, all fees are still in the sink.
-		require.EqualValues(t, 2000, postsink-presink)
-		require.EqualValues(t, 0, postprop-preprop)
+		// Rest of the tests only make sense with payout active
+		if ver < payoutsBegin {
+			return
+		}
 
-		// Do the next block, which moves part of the fees + bonus to proposer
-		dl.fullBlock()
-		postsink = micros(dl.t, dl.generator, genBalances.FeeSink)
-		postprop = micros(dl.t, dl.generator, proposer)
-		t.Log(" postsink2", postsink)
-		t.Log(" postprop2", postprop)
+		// Get the feesink down low, then drain it by proposing.
```

Comment:
The fee sink should go to zero here?

Reply:
No, the final, repetitive lines show we don't let it go below MinBalance.
```
		dl.beginBlock()
		dl.endBlock(proposer)
		require.EqualValues(t, micros(t, dl.generator, feesink), 100_000)

		dl.beginBlock()
		dl.endBlock(proposer)
		require.EqualValues(t, micros(t, dl.generator, feesink), 100_000)
```